compaction/compaction_manager.cc

```diff
@@ -468,8 +467,9 @@ class sstables_task_executor : public compaction_task_executor, public sstables_
         set_sstables(std::move(sstables));
     }
 
-    virtual ~sstables_task_executor();
+    virtual ~sstables_task_executor() = default;
 
+    virtual void release_resources() override;
```

Comment:
nit: let's declare it a noexcept since it's called from destructor (of the deferred action)

Otherwise, LGTM